### PR TITLE
NMS-13792: Upgrade to Antora 3.1.1

### PR DESCRIPTION
**Pull request by indigo423**
This PR migrates from Antora 2.3.4 to 3.1.1. With Antora 3 there is now an extension framework and the Lunr plugin is migrated and is now implemented as an extension. The Lunr extension got some fundamental rewrite and it required us to rebuild our search integration in our Antora UI

## Reviewers hint
We did a modification in the search.js to highlight search results in NMS-13540 which got busted. To get it back to work, I had to use the original file from their repository. For sustainability, we probably should look into getting the contribution [upstream](https://gitlab.com/antora/antora-lunr-extension) instead of maintaining it as a fork.

The customizations for admonitions, external Javascript for MathJAX are all moved now to the Antora UI and we don't need to maintain it per repository, i.e. cloud docs and docs.opennms.com.

The supplemental_ui directory needs to be renamed to match the Antora Lunr extension, which expects the directory to be existent. Additionally, the supplemental UI components for Admonitions and some CSS updates are moved to the Antora UI to be globally available instead of per-project.

The [site.zip](https://output.circle-artifacts.com/output/job/6098c568-d784-4310-a333-0361038d5249/artifacts/0/site.zip) can be used for local testing.

## Risk

If we break it, we can revert this PR and fix it :)

JIRA: https://issues.opennms.org/browse/NMS-13792

### CODE REVIEW

**Review by Bonrob2 — COMMENTED** [review]
makes sense to me, but unable to test it.

**Comment by indigo423** [comment]
> makes sense to me, but unable to test it.

yes you can test it 

1. Download the ZIP file from the last build, like here: https://1666-251686651-gh.circle-artifacts.com/0/site.zip
2. Extract it
3. Open the index.html in your browser

Report issues if you find some :)

**Comment by indigo423** [comment]
Instead of 29 you see now "latest" as part of the URL for example.

**Comment by RangerRick** [comment]
So I assume before we merge this, we need to update the redirects to send you to `latest` instead of the short version.

**Comment by indigo423** [comment]
> So I assume before we merge this, we need to update the redirects to send you to `latest` instead of the short version.

yes, before or after the merge

**Comment by Bonrob2** [comment]
> > makes sense to me, but unable to test it.
> 
> yes you can test it
> 
>     1. Download the ZIP file from the last build, like here: https://1666-251686651-gh.circle-artifacts.com/0/site.zip
> 
>     2. Extract it
> 
>     3. Open the index.html in your browser
> 
> 
> Report issues if you find some :)

Looks good to me! It will be great to cross-reference between docs and components.

**Comment by indigo423** [comment]
@GanimanSwift Just pulling you in here for this PR which introduces the "latest" floating tag instead of the version numbers "29" for Horizon and "2021". Do you think this is a showstopper? We would need to adjust our redirect regexp. If you think there is too much risk with it, I can easily remove it and make a dedicated PR with it. What do you think?

**Comment by Bonrob2** [comment]
@GanimanSwift are you able to address @indigo423's question? I'd like to merge this if possible.

**Comment by RangerRick** [comment]
Ah so if it does "latest" it doesn't also do one based on the version?

**Comment by indigo423** [comment]
> Ah so if it does "latest" it doesn't also do one based on the version?

Yes for Horizon it would only be only one version, right now we could show 29.0.6 and the URL path would have `latest` instead of 29.0.6, 29, 30 or whatever our last release is.

For Meridian, latest would point to the most current Meridian 2021 release. All the older versions have the version number in the path.

For all other projects, it would be the same. There are two exceptions, ALEC and OpenNMS-JS, we publish SNAPSHOT releases. They would get the `latest` floating tag. We would need to remove these SNAPSHOT docs to have `latest` to be consistently the last release we want to encourage people to run and to upgrade to.

**Review by fooker — DISMISSED** [review]
I can try to bring it back/bring it upstream. Should we make this a separate issue or is it a stopper for this?

**Comment by indigo423** [comment]
> I can try to bring it back/bring it upstream. Should we make this a separate issue or is it a stopper for this?

I don't think we should make it a show stopper for this PR. We can reopen NMS-13540 and track it there?

**Comment by Bonrob2** [comment]
@indigo423 are we able to merge this now? Chris has finished the work on his side.

**Review by RangerRick — DISMISSED** [review]
yeah, I like not doing the `latest` on the antora side, makes more sense to do it with a symlink as part of the deploy

**Comment by Bonrob2** [comment]
@indigo423 can we merge this now, or do we need to do additional work?

**Comment by Bonrob2** [comment]
@indigo423 or @RangerRick ... doing some clean up and noticed this is still sitting around. Could we move forward and upgrade Antora?

**Comment by Bonrob2** [comment]
I have re-opened NMS-13540 to request the Lunr highlighting work be done upstream. IT has done the regex work. Could we merge this PR? @indigo423?

File: antora-playbook.yml
Comment (indigo423) [inline antora-playbook.yml]: We have a dependency on the Antora UI update here: https://github.com/opennms-forge/antora-ui-opennms/pull/27

⚠️ This is a placeholder here that links to a CircleCI build artifact for testing.

**Comment by indigo423** [comment]
Released a new version with the Antora UI and used it here in the repository instead of the build artifact.

**Review by Bonrob2 — COMMENTED** [review]
LGTM, but will defer to Ben for approval.

**Review by RangerRick — APPROVED** [review]
assuming the folks who are affected by it (product? support?) are ok with search hilighting going away, I'm OK with this

**Pull request merged**
